components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -343,6 +345,7 @@ class SystemEngineSession(
         val webSettings = webView.settings
         webSettings.userAgentString = toggleDesktopUA(webSettings.userAgentString, enable)
         webSettings.useWideViewPort = enable
+        defaultSettings?.useWideViewPort = enable
```

Comment:
I'm concerned about the interaction between `useWideViewPort` and `toggleDesktopMode` here.

That desktop mode internally flips `useWideViewPort` is an implementation detail and ideally shouldn't be leaked to the consumer. Or in other words I think we shouldn't let the consuming app deal with figuring out how to not override itself.

As a consumer I would expect the following behavior:
- If I do not specify `useWideViewPort` then desktop mode will toggle it on/off.
- If I do specify `useWideViewPort = false` then desktop mode will need to toggle it on/off
- If I do specify `useWideViewPort = true` then I expect it to be always on and desktop mode will never toggle it off.

What do you think? (CC @csadilek)

This will probably require the addition of a boolean flag (instead of only writing/reading from WebSettings) so that we can figure out whether this WebSetting was flipped deliberately by `useWideViewPort` (and needs to stay that way) or by something else like desktop mode.

Reply:
@pocmo @psymoon yes, that sounds good! I think the user supplied default setting (which is for all sessions) shouldn't change by toggling desktop mode. It should just always reflect what the user configured. Of course, when "asking" a specific session for `session.setttings.useWideViewPort` the result could be different than the default if the session is in desktop mode or it was configured directly. We can add some session-internal state to decide if we have to toggle `useWideViewPort` when `toggleDesktopMode` is invoked, as @pocmo suggested, based on the value of `settings.useWideViewPort` before desktop mode is toggled.

Reply:
Doesn't the desktop mode also require non-mobile UA? At least in FFTV, `useWideViewPort = true` doesn't guarantee desktop mode 🤔 . I agree with your concern though - maybe @dnarcese can provide additional context on the consumer behaviour.

Reply:
Yeah, currently `toggleDesktopMode` does enable widemode AND changes the UA.

Reply:
>    -  If I do not specify useWideViewPort then desktop mode will toggle it on/off.
>    - If I do specify useWideViewPort = false then desktop mode will need to toggle it on (and off again)
>    - If I do specify useWideViewPort = true then I expect it to be always on and desktop mode will never toggle it off.

I agree with this behaviour.